### PR TITLE
Pyapp

### PR DESCRIPTION
**Pull request by vishalkor**
Hi Sunil,
In this branch i have added python appium script along with .gitignore and test_requirement.txt file.
Please review the same.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,11 +4,16 @@
...
 *.pyc
 *.swp
 *.swo
 *.pyo
+__*__
+Results/
+.html
+.cache/
+assets/
+*.log
```
Comment (ersuniljoshi) [inline .gitignore]: New line

File: conftest.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    extra = getattr(report, 'extra', [])
+    if report.when == 'call':
+        xfail = hasattr(report, 'wasxfail')
+        if (report.skipped and xfail) or (report.failed and not xfail):
+                if os.path.exists("myfile1"):
+                        file = open('myfile1', 'r')
+                        screenshot = file.read()
+                        os.remove("myfile1")
+                        extra.append(pytest_html.extras.image(screenshot, 'Screenshot'))
+        report.extra = extra
```
Comment (ersuniljoshi) [inline conftest.py]: New line

File: library/element_codexes.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            return expected_conditions.presence_of_element_located(
+            (By.ID, codex_details[1]))
+        elif codex_details[0] == "class":
+            return expected_conditions.presence_of_element_located(
+            (By.CLASS_NAME, codex_details[1]))
+        elif codex_details[0] == "xpath":
+            return expected_conditions.presence_of_element_located(
+            (By.XPATH, codex_details[1]))
+        else:
+            return False
```
Comment (ersuniljoshi) [inline library/element_codexes.py]: New line

File: createSurvey_using_appium.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    else:
+        logger.info("Added matrix rating type of question")
+
+
+    # Verify survey in preview
+    logger.info("Verifying survey is created......")
+    rv = QB.previewAndTestSurvey()
+    if not rv:
+        logger.info("Unable preview and test survey")
+        QB.tearDown()
```
Comment (ersuniljoshi) [inline createSurvey_using_appium.py]: QB.tearDown(), we are calling this from if and else both, please move it common place.

File: library/question_types.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        rv = self.waitForElement("tripleDotButtonForPreviewAndTest")
+        if not rv:
+            return rv
+        ele = self.getWebElement("tripleDotButtonForPreviewAndTest")
+        ele.click()
+        ele = self.getWebElement("selectPreviewAndTestButton")
+        ele.click()
+        return True
+
+
```
Comment (ersuniljoshi) [inline library/question_types.py]: Remove extra blank lines.

File: createSurvey_using_appium.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    logger.info("Executing Testcase 1...........")
+    logger.info("Signing in..........")
+
+    # User sign in
+    rv = QB.userSignIn()
+    assert rv
+
+    # create survey
+    rv = QB.createSurvey("survey 1")
+    if rv == False:
```
Comment (ersuniljoshi) [inline createSurvey_using_appium.py]: Variable is boolean itself.

File: .gitignore (hunk 1 above)
Comment (vishalkor) [inline .gitignore]: Newline Added.

File: conftest.py (hunk 2 above)
Comment (vishalkor) [inline conftest.py]: Newline Added.

File: library/element_codexes.py (hunk 3 above)
Comment (vishalkor) [inline library/element_codexes.py]: Newline Added.

File: library/question_types.py (hunk 5 above)
Comment (vishalkor) [inline library/question_types.py]: Removed extra lines.

File: createSurvey_using_appium.py (hunk 6 above)
Comment (vishalkor) [inline createSurvey_using_appium.py]: if rv == False is replaced by if not rv

File: createSurvey_using_appium.py (hunk 4 above)
Comment (vishalkor) [inline createSurvey_using_appium.py]: Done.

**Pull request merged**